src/KmerIndex.cpp

```diff
@@ -1096,9 +1096,12 @@ void KmerIndex::match(const char *s, int l, std::vector<std::pair<KmerEntry, int
           auto search2 = kmap.find(rep2);
           bool found2 = false;
           int  found2pos = pos+dist;
-          if (search2 == kmap.end()) {
```

Comment:
Check if `search2` result is empty! and then raise the flag `found2` to be true!
That's not logical.